ods_ci/tests/Resources/OCP.resource

```diff
@@ -338,6 +338,33 @@ Get Resource Attribute
     END
     RETURN    ${value}
 
+Wait For Installplan And Approve It
+    [Documentation]    Wait for the operator's installplan to appear then approve it in case of Manual installplan approval
+    [Arguments]    ${namespace}    ${operator_name}     ${subscription_name}    ${operator_version}=${EMPTY}
+    ${csv} =    Evaluate    '${operator_name}.${operator_version}' if '${operator_version}' else '${operator_name}'
+    Log    The csv is: ${csv}, operator_version: ${operator_version}, operator_name: ${operator_name}, subscription_name: ${subscription_name}     console=yes
+    Wait Until Keyword Succeeds    10m    5s    Search Install Plan    ${csv}    ${namespace}
+    ${installplan_name}=    Run   oc get subscription ${subscription_name} -n ${namespace} -o json | jq '.status.installplan.name'
+    IF    "${operator_version}" != "${EMPTY}"
+        ${installplan_version_exist}=    Run And Return Rc   oc get installplan -n ${namespace} | grep "${operator_version}"
+        IF    ${installplan_version_exist} != 0
+            ${installplans}=    Run   oc get installplan -n ${namespace}
+            FAIL    Failed to find installplan for operator: ${operator_name} version: ${operator_version}\nExisting installplans:\n${installplans}
+        END
+    END
+    ${installplan_approval}=     Get Resource Attribute      ${namespace}
+    ...      InstallPlan       ${installplan_name}        .spec.approval
```

Comment:
you're using the variable `installplan_name` which is assigned before checking the existing install plans when `operator_version` arg is passed. 
How do you ensure that the right install plan name is used when multiple install plans exist _(in case `operator_version` arg is passed)_?

Reply:
yeah but at this point, wouldn't make more sense to check that the install-plan which is referred in the subscription has the expected operator version?
I think this check is a bit too loose: just check that one of the install plans have the given operator version, although we know already which is the install plan used by the operator.

You know the install plan? Check just that one no?

Reply:
It is because we do not always have the operator version.
But maybe I do not understand your point.
Could you please write your suggestion?

Reply:
Updated

Reply:
ty